lib/spandx/php/parsers/composer.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Spandx
+  module Php
+    module Parsers
+      class Composer < ::Spandx::Core::Parser
+        def self.matches?(filename)
+          File.basename(filename) == 'composer.lock'
+        end
+
+        def parse(file_path)
+          items = Set.new
+          composer_lock = JSON.parse(IO.read(file_path))
+          composer_lock['packages'].concat(composer_lock['packages-dev']).each do |dependency|
+            items.add(Spandx::Core::Dependency.new(
+                        name: dependency['name'],
+                        version: dependency['version'],
+                        licenses: catelogue_licenses(dependency['license']),
+                        meta: dependency
+                      ))
+          end
+          items
+        end
+
+        private
+
+        def catelogue_licenses(licenses)
+          licenses.map do |license_name|
+            found_license = catalogue[license_name]
+            Spandx.logger.info("Could not find SPDX license id for #{license_name}") if found_license.nil?
```

Comment:
one if the license is "BSD"  but there is not such license in "BSD" in SPDX index 

this is the dependency that has "BSD"  https://github.com/hamcrest/hamcrest-php

Reply:
Don't worry about searching for the licenses from this class. I've started to consolidate the license detection in one place. Details in https://github.com/mokhan/spandx/pull/8. For now it might be easier to specify a blank list of licenses.

.ie.

```ruby
            items.add(Spandx::Core::Dependency.new(
                        name: dependency['name'],
                        version: dependency['version'],
                        licenses: [],
                        meta: dependency
                      ))
```

This will reduce the merge conflicts and allow you to focus on the parsing logic. WDYT?